packages/common/src/store/transactions.slice.ts

```diff
@@ -94,15 +93,17 @@ export const getNoncesInQueue = createSelector(
 
 export const hasNonceConflictInQueue = createSelector(getNoncesInQueue, (nonces) => nonces > 1);
 
-export const hasNonceConflict = (account: TAddress, chainId: number, nonce: BigNumberish) =>
-  createSelector(
-    getApprovedTransactions,
-    (transactions) =>
-      transactions.find(
-        (h) =>
-          h.tx.from === account && h.tx.chainId === chainId && bigify(h.tx.nonce).eq(bigify(nonce))
-      ) !== undefined
-  );
+export const hasNonceConflict = createSelector(
+  getApprovedTransactions,
+  getCurrentTransaction,
+  (transactions, currentTx) =>
+    transactions.find(
+      (h) =>
+        h.tx.from === currentTx.tx.from &&
+        h.tx.chainId === currentTx.tx.chainId &&
+        bigify(h.tx.nonce).eq(bigify(currentTx.tx.nonce))
```

Comment:
Why is bigify necessary here? Can the nonce be in a different format?

Reply:
Just to be safe as it is a Redux value it may be serialized, right?